crutest/src/main.rs

```diff
@@ -2185,11 +2351,149 @@ async fn perf_workload(
+        RandReadWriteMode::Write => "write",
+    };
+
+    if cfg.fill {
+        println!("filling region before {desc}");
+        fill_workload(guest, ri, true).await?;
+    }
+
+    if cfg.blocks_per_io > ri.total_blocks {
+        bail!("too many blocks per IO; can't exceed {}", ri.total_blocks);
+    }
+
+    println!(
+        "\n-----------------------------------\
+        \nrandom {desc} with {} chunks ({} block{})",
+        human_bytes((cfg.blocks_per_io as u64 * ri.block_size) as f64),
+        cfg.blocks_per_io,
+        if cfg.blocks_per_io > 1 { "s" } else { "" },
+    );
+    print_region_description(ri, cfg.encrypted);
+
+    let stop = Arc::new(AtomicBool::new(false));
+    let byte_count = Arc::new(AtomicUsize::new(0));
+
+    let block_size = ri.block_size as usize;
+    let total_blocks = ri.total_blocks;
+    for _ in 0..cfg.io_depth {
+        let stop = stop.clone();
+        let guest = guest.clone();
+        let byte_count = byte_count.clone();
+        tokio::spawn(async move {
+            match cfg.mode {
+                RandReadWriteMode::Write => {
+                    let mut buf = BytesMut::new();
+                    buf.resize(cfg.blocks_per_io * block_size, 0u8);
+                    let mut rng = rand_chacha::ChaCha8Rng::from_entropy();
+                    rng.fill_bytes(&mut buf);
+                    let block_shift = block_size.trailing_zeros();
+                    while !stop.load(Ordering::Acquire) {
+                        let offset =
+                            rng.gen_range(0..=total_blocks - cfg.blocks_per_io);
+                        guest
+                            .write(
+                                Block::new(offset as u64, block_shift),
+                                buf.clone(),
+                            )
+                            .await
+                            .unwrap();
+                        byte_count.fetch_add(buf.len(), Ordering::Relaxed);
+                    }
+                }
+                RandReadWriteMode::Read => {
+                    let mut buf = Buffer::new(cfg.blocks_per_io, block_size);
+                    let mut rng = rand_chacha::ChaCha8Rng::from_entropy();
+                    let block_shift = block_size.trailing_zeros();
+                    while !stop.load(Ordering::Acquire) {
+                        let offset =
+                            rng.gen_range(0..=total_blocks - cfg.blocks_per_io);
+                        guest
+                            .read(
+                                Block::new(offset as u64, block_shift),
+                                &mut buf,
+                            )
+                            .await
+                            .unwrap();
+                        byte_count.fetch_add(buf.len(), Ordering::Relaxed);
+                    }
+                }
+            }
+            Ok::<(), CrucibleError>(())
+        });
+    }
+
+    let mut prev = 0;
```

Comment:
I don't think it matters really for the ways we will be using this test, but the first
second of time elapsed, if I'm reading this correctly, will count not only all the IO
that happened in that first second, but also include all the IO generated duing the setup
loop as well.  Basically all the IO done by tasks `0..cfg.io_depth` as we spun them out.

It probably only going to make a difference if you had a large number of tasks with a
short test duration.  Given the default of 60 seconds, it's probably not going to move
the needle much.

Reply:
Yeah, I added some code to skip the first data sample (af4b3aa2d33e445425d1e899f59998ebf0c7a75a)